libyul/AsmAnalysis.cpp

```diff
@@ -485,18 +487,38 @@ bool AsmAnalyzer::operator()(ForLoop const& _for)
 	if (!expectExpression(*_for.condition))
 		success = false;
 	m_stackHeight--;
+
+	m_info.stackHeightInfo[&_for] = m_stackHeight - scope(&_for.pre).numberOfVariables();
```

Comment:
Why not assign this before the pre block is visited?

Reply:
Actually - this is re-assigned below. Should either remove one of the assignments or turn the other one into an assertion.